extension/chrome/elements/attachment_preview.ts

```diff
@@ -0,0 +1,78 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+'use strict';
+
+import { Att } from '../../js/common/core/att.js';
+import { AttachmentDownloadView } from './attachment.js';
+import { BrowserMsg } from '../../js/common/browser/browser-msg.js';
+import { Catch } from '../../js/common/platform/catch.js';
+import { KeyStore } from '../../js/common/platform/store/key-store.js';
+import { PgpMsg } from '../../js/common/core/pgp-msg.js';
+import { View } from '../../js/common/view.js';
+import { Xss } from '../../js/common/platform/xss.js';
+import { Ui } from '../../js/common/browser/ui.js';
+
+type AttachmentType = 'img' | 'txt';
+
+View.run(class AttachmentPreviewView extends AttachmentDownloadView {
+  private attachmentPreviewContainer = $('#attachment-preview-container');
+
+  constructor() {
+    super();
+  }
+
+  public render = async () => {
+    try {
+      Xss.sanitizeRender(this.attachmentPreviewContainer, `${Ui.spinner('green', 'large_spinner')}<span class="download_progress"></span>`);
+      this.att = new Att({ name: this.origNameBasedOnFilename, type: this.type, msgId: this.msgId, id: this.id, url: this.url });
+      await this.downloadDataIfNeeded();
+      const result = this.isEncrypted ? await this.decrypt() : this.att.getData();
+      if (result) {
+        const blob = new Blob([result], { type: this.type });
+        const url = window.URL.createObjectURL(blob);
+        const attachmentType = this.getAttachmentType(this.origNameBasedOnFilename);
+        const downloadBtn = $(`<a href="${url}" download="${Xss.escape(this.origNameBasedOnFilename)}" class="download-attachment">
```

Comment:
This looks like it's partially recreating `Browser.saveToDownloads`. Does this version work differently? Does it work better? Then we could consider replacing contents of `saveToDownloads` and using that here, too?

Or if it's just different, maybe we could put both to `Browser` class and document the difference.